### ISSUE TITLE
Maximum number of intrinsics

### ISSUE THREAD

**Issue by dalnoguer** [openMVG/openMVG#1668]
Hello!

I am trying to run sfm pipeline with different intrinics. In my setting I have 30 different cameras so I would like to assign different intrinsics to all of them. However, the pipeline only allows me to set at most 3 different intrinsics.

Thanks :D

**Comment by pmoulon** [comment]
What do you mean it allows you to set only 3 intrinsics?
- You can export the sfm_data.json file and then customize it with your own code (Python, C++, ...) and use as many intrinsic as you want.

If you mean that you have 30 images and you want to keep separate intrinsic for each camera, you can use this option `----group_camera_model 0` to do not group camera that have the same intrinsic model and value https://github.com/openMVG/openMVG/blob/develop/src/software/SfM/main_SfMInit_ImageListing.cpp#L182

**Comment by dalnoguer** [comment]
Sorry maybe I did not make myself clear. My problem is: 

I have 1469 frames in a directory: 1443 are coming from a moving camera and 26 are coming from fixed cameras. I would like to assign all the 1443 to one intrinsics group and the other 26 have each one different intrinsics. All the intrinsics are known and I wish them to be fixed during the whole sfm pipeline. 

How I am doing in at the moment is I call:

```
    pIntrisics = subprocess.Popen(
        [os.path.join(OPENMVG_SFM_BIN, "openMVG_main_SfMInit_ImageListing"), "-i", input_dir, "-o", matches_dir, "-d",
         camera_file_params, "-c", "3", "-k", "1399.27;0;1106.26;0;1399.27;601.167;0;0;1"])
```

And then I edit the sfm_data.json in the matches directory in the following way:

I create an intrinsics list that contains the different 27 intrinsics and a corresponding key from 0 to 26. Then I change the views list so that each view intrinsics_id points to the correct intrinsics. Am I missing something?

Thanks:D

**Comment by pmoulon** [comment]
Yes you are doing it right. To fix them, you will have to set the flag `--refineIntrinsics NONE` in the SFM phase.

**Comment by pmoulon** [comment]
Can you share the final JSON file you generated, I will perhaps notice some ID conflict.

I would recommend you to generate the sfm_data with ungrouped camera (see `-g 0` option) and then use the same ID for your views. This way no conflict should happen and the intrinsic that are not used will just be ignored by the syste,.

**Comment by dalnoguer** [comment]
I reply here since I can't attach a json file.

Okay I tried generating the sfm_data with ungrouped camera and then
modifying just the first 27 intrinsics and then adjusting for each view the
intrinsics group. That is the resulting sfm_data json:

Best,

Dal

**Comment by pmoulon** [comment]
Did it work as expected this way (no crash)?

**Comment by dalnoguer** [comment]
No, but I found the solution. Apparently all the intrinsics listed in
sfm_data.json need to be assigned to at least a view. Otherwise the next
error shows up:

There is orphan intrinsics data or poses (do not depend on any view)

Thanks a lot for your help,

Best,

Dal
